envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactory.kt

```diff
@@ -50,15 +49,11 @@ class RBACFilterFactory(
             }
 
             val clients = resolveClients(incomingEndpoint, incomingPermissions.roles)
-            val policyName = clients.joinToString(",")
-
             val principals = clients.flatMap { mapClientToPrincipals(it, snapshot) }
```

Comment:
I think we can do better. We still compute `mapClientToPrincipals` multiple times for the same policyName. I still propose below solution:

```kotlin
            val policy: Policy.Builder = clientToPolicyBuilder.computeIfAbsent(policyName) {
                Policy.newBuilder().addAllPrincipals(
                    clients.flatMap { mapClientToPrincipals(it, snapshot) }
                )
            }
            
            if (policy.principalsList.isNotEmpty()) {
                val combinedPermissions = createCombinedPermissions(incomingEndpoint)
                policy.addPermissions(combinedPermissions)
            }
```

Reply:
Your version causes errors, I need to figure out why.
```
[2020-04-24 07:50:58.660][15][debug][misc] [source/common/protobuf/utility.cc:254] Proto validation error; throwing Proto constraint validation failed (RBACValidationError.Rules: ["embedded message failed validation"] | caused by RBACValidationError.Policies[key]: ["emb
edded message failed validation"] | caused by PolicyValidationError.Permissions: ["value must contain at least " '\x01' " item(s)"]): rules {
```

Reply:
OK, so I think we can keep it as it is now